Automattic/wp-calypso
# Checkoutv2: Refactor sidebar promotions into CheckoutSidebarNudge

This PR consolidates a number of separate sidebar promotional components into one 'CheckoutSidebarNudge' wrapper component. There are a number of components that should be refactored, but this PR specifically handles plan interval upsells - see here for full list https://github.com/Automattic/payments-shilling/issues/2422#issuecomment-1965647926

Part of these changes include changing the upsell component from a CTA button to a word bubble with an inline text link CTA:

| v1 | v2 |
| ----- | ----- |
| ![image](https://github.com/Automattic/wp-calypso/assets/16580129/d69383ae-7d73-4eee-bfc4-139799102c62) | ![image](https://github.com/Automattic/wp-calypso/assets/16580129/e01a45a8-8fb8-4ec3-81dd-8d62a2bf1f7b) |

| v1 | v2 |
| ----- | ----- |
| ![image](https://github.com/Automattic/wp-calypso/assets/16580129/53089d5c-36f0-427a-a1f5-2043d69f2f91) | ![image](https://github.com/Automattic/wp-calypso/assets/16580129/9315e8d6-048c-4850-a2fb-826570a73621) |

Related to https://github.com/Automattic/payments-shilling/issues/2422

#### Note - I intend to merge this for both v1 and v2, since the sidebar wrapper is already in production and these changes will benefit v1 as well

## Proposed Changes

* Update the `PromoCard` component to a newer `CheckoutPromoCard` which can handle multiple upgrade levels
* Move the monthly -> annual upsell from the Features List to the new checkout word bubble
* Update the `onUpgradeClick` function to handle more than annual -> biennial upgrades
* Styling changes to the word bubble in v2 design along with handling style regressions of other sidebar components such as the ones found in `SecondaryCartPromotions`
* Change `CheckoutSidebarNudgeWrapper` to flex and remove margin from nudge components

## Testing Instructions



Since the onUpgradeClick function has been modified for both checkout v1 (currently in production) and v2, we'll need to check that the function works as expected in both checkout flows - so the following tests need to be done for both v1 and v2 of checkout

**Test DotCom plans**
* Add a monthly plan to your cart and go to checkout - you should see an upsell nudge that reads `Longer plan billing cycles save you money and include a custom domain for free for the first year.`
* Add a monthly plan and a domain, you should be prompted to switch to an annual plan to receive your domain free for the first year (including your domain being listed in the sidebar) `Save money with longer billing cycles and get %(domainMeta)s free for the first year by {{upgradeLink}}switching to an annual plan.{{/upgradeLink}}`
* Click on sidebar upgrade link to switch to annual plan
* Now that you have an annual plan in the cart you should see the following nudge `Save up to %(savingsPercentage)s% on longer billing cycles! It's hassle-free and easy on your wallet. {{upgradeLink}}Switch to a two-year plan and save.{{/upgradeLink}}`, click on sidebar upgrade link to switch to two year plan
* At this stage we stop showing upgrade nudges for biennial -> triennial plans terms.

**Test Jetpack and Akismet Plans**
* Jetpack and Akismet do not use the new word bubble upsell yet, so at this time please add a Jetpack or Akismet product and ensure the old nudge is still showing as expected

**DIFM**
* Go to `http://calypso.localhost:3000/start/do-it-for-me/new-or-existing-site` and continue through the onboarding steps until you reach checkout
* You should see in the sidebar the DIFM `What's next?` stepper and no other upsell nudges

**Upcoming Renewals**
* This one is trickier, you'll need a site with multiple purchases from around the same date with similar expirations
* When you go to manually renew one of your purchases, Checkout should prompt you renew the others
